File: menu-scanner-frontend-owner/src/features/platform-bank/store/models/request/platform-bank-request.ts
export interface PlatformBankRequestModel {
  name: string;
  accountName: string;
  accountNumber: string;
  description?: string;
  status?: string;
  displayOrder?: number;
  image?: {
    sm?: string;
    md?: string;
    o?: string;
  } | null;
}

export interface AllPlatformBankRequestModel {
  search?: string;
  status?: string;
  pageNo?: number;
  pageSize?: number;
}

File: menu-scanner-backend/src/main/java/com/emenu/features/master/dto/request/PlatformBankRequest.java
package com.emenu.features.master.dto.request;

import com.emenu.enums.common.Status;
import com.emenu.shared.dto.ImageUrls;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlatformBankRequest {

    @NotBlank(message = "Bank name is required")
    private String name;

    @NotBlank(message = "Account name is required")
    private String accountName;

    @NotBlank(message = "Account number is required")
    private String accountNumber;

    private String description;
    private Status status;
    private Integer displayOrder;
    private ImageUrls image;
}
